pymongo/cursor.py

```diff
@@ -1277,23 +1287,35 @@ def __deepcopy__(self, memo: Any) -> Any:
         """
         return self._clone(deepcopy=True)
 
-    def _deepcopy(self, x, memo=None):
+    @overload
+    def _deepcopy(self, x: Iterable, memo: Optional[Dict[int, Union[List, Dict]]] = None) -> List:
```

Comment:
Can these be `Mapping` instead of `Dict`?

Reply:
I don't think so because `copy.deepcopy` takes in a `Dict` and we call that on line's 1325 and 1331

Reply:
👍🏼